docs/useDataProvider.md

```diff
@@ -48,12 +48,56 @@ But the recommended way to query the Data Provider is to use the dataProvider me
 
 **Tip**: The `dataProvider` returned by the hook is actually a *wrapper* around your Data Provider. This wrapper logs the user out if the dataProvider returns an error, and if the authProvider sees that error as an authentication error (via `authProvider.checkError()`).
 
-**Tip**: If you use TypeScript, you can specify a record type for more type safety:
+## TypeScript
+
+The `useDataProvider` hook accepts a generic parameter for the `dataProvider` type. This is useful when you added custom methods to your `dataProvider`:
+
+```tsx
+// In src/dataProvider.ts
+import { DataProvider } from 'react-admin';
+
+export interface CustomDataProviderMethods extends DataProvider {
```

Comment:
The interface name is misleading. `AugmentedDataProvider` or `DataProviderWithCustomMethods` would be better.